cmd/lk/main.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/urfave/cli/v3"
+
+	livekitcli "github.com/livekit/livekit-cli"
+	"github.com/livekit/protocol/logger"
+	lksdk "github.com/livekit/server-sdk-go/v2"
+)
+
+func main() {
+	app := &cli.Command{
+		Name:                  "lk",
+		Usage:                 "CLI client to LiveKit",
+		Description:           "A suite of command line utilities allowing you to access LiveKit APIs services, interact with rooms in realtime, and perform load testing simulations.",
+		Version:               livekitcli.Version,
+		EnableShellCompletion: true,
+		Suggest:               true,
+		HideHelpCommand:       true,
+		Flags: []cli.Flag{
+			&cli.BoolFlag{
+				Name: "verbose",
+			},
+		},
+		Commands: []*cli.Command{
+			{
+				Name:   "generate-fish-completion",
+				Action: generateFishCompletion,
+				Hidden: true,
+				Flags: []cli.Flag{
+					&cli.StringFlag{
+						Name:    "out",
+						Aliases: []string{"o"},
+					},
+				},
+			},
+		},
+		Before: func(ctx context.Context, cmd *cli.Command) error {
+			logConfig := &logger.Config{
+				Level: "info",
+			}
+			if cmd.Bool("verbose") {
+				logConfig.Level = "debug"
+			}
+			logger.InitFromConfig(logConfig, "livekit-cli")
+			lksdk.SetLogger(logger.GetLogger())
+
+			return nil
+		},
+	}
+
+	app.Commands = append(app.Commands, TokenCommands...)
```

Comment:
I've seen a pretty convenient CLI pattern in other projects:

```go
// main.go
var App = &cli.Command{
		Name:                  "lk",
		// ...
}

func main() {
	if err := app.Run(context.Background(), os.Args); err != nil {
		fmt.Println(err)
	}
}
```

```go
// some-sub-command.go

func init() {
	cmd := &cli.Command{
		Name:                  "some-sub-cmd",
		// ...
	}
	App.Commands = append(App.Commands, cmd)
}
```

That way each CLI sub-command file can register its own list of commands without the need to pile all registrations in main.

Reply:
Only issue with that is that order matters. If commands self-register, the ordering in Usage is at the whims of the compiler. I'd rather have control over that, even if it's a little awkward.